### PR TITLE
add github ISSUE_TEMPLATE files

### PR DESCRIPTION
**Pull request by cliffordfajardo**
## What is the change?
1. Update `bug_report.md` to `bug-report.yml` to enable Github's form based issue template
   - https://youtu.be/qQE1BUkf2-s?t=23
2.  update `config.yml` file to help direct users to the helpful pages

## Motivation
- encourage's bug reporter's to put more care into their bug report before submission
- this may help maintainer's receive more detailed & higher quality bug report's
- adds helpful tips for user's during the process of creating a bug/issue report

## Demo of Change
- this PR is similar to this one I created for `remix-auth` repo: https://github.com/sergiodxa/remix-auth/pull/87

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/config.yml [hunk 1]
```diff
@@ -1,5 +1,8 @@
 blank_issues_enabled: false
 contact_links:
+  - name: 🤔 Long question or idea
+    url: https://github.com/pedroslopez/whatsapp-web.js/discussions
+    about: Ask long-form questions and discuss ideas.
```
Comment (PurpShell) [inline .github/ISSUE_TEMPLATE/config.yml:5]:
```suggestion
```

Update it so that we keep everything question related on Discord..

**Review by purpshell — CHANGES_REQUESTED** [review]
looks great, just one change

**Review by purpshell — APPROVED** [review]
lgtm

**Comment by tuyuribr** [comment]
Sounds like a automated tool, I don't feel like approving this PR. Also there is dead links in the bug template

**Comment by cliffordfajardo** [comment]
Hi @tuyuribr - no this is not autogenerated by script  
 I'm the original creator of this pull request template & I am a real user of the repository😄 

If I'm using a repository repo which I'm personally using and I see that there is no `config.yml` or `bug_report.yml`, I create a pull request because I believe it useful to have especially if people decide to report a bug on your repo


I've contributed a similar PR template to popular projects like `react-query` 35k+ stars, solidJS 15k + stars etc.
- https://github.com/tannerlinsley/react-query/blob/master/.github/ISSUE_TEMPLATE/bug_report.yml
- https://github.com/solidjs/solid/blob/main/.github/ISSUE_TEMPLATE/config.yml

The intent is just to help maintainers out a little bit through with the relatively new github `.yml` files
- https://youtu.be/qQE1BUkf2-s?t=23

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
+name: '🐛 Bug report'
+description: Create a report to help us improve
+body:
+  - type: markdown
+    attributes:
+      value: |
+        Thank you for reporting an issue :pray:.
+
+        This issue tracker is for reporting bugs found in `whatsapp-web.js` (https://github.com/pedroslopez/whatsapp-web.js).
+        If you have a question about how to achieve something and are struggling, please post a question
```
Comment (cliffordfajardo) [inline .github/ISSUE_TEMPLATE/bug_report.yml:10]: @PurpShell @pedroslopez - you might need to enable github discussions for your repository for the discussions tab
- https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/enabling-or-disabling-github-discussions-for-a-repository

**Comment by tuyuribr** [comment]
@cliffordfajardo 

Sorry bro, my bad. hahahaha, I thought you were a bot because it had links to the discussion page but we only use the discord channel to do these kinds of things. Thanks for your great work, and I hope to drink beer with you in the future 🍻🍻

**Comment by cliffordfajardo** [comment]
> Sorry bro, my bad. hahahaha, I thought you were a bot because it had links to the discussion page but we only use the discord channel to do these kinds of things. Thanks for your great work, and I hope to drink beer with you in the future 🍻🍻

Cheers mate 🍺🍻

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (PurpShell) [reply .github/ISSUE_TEMPLATE/bug_report.yml:10]: i don't think we should do that, again.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        inside of `whatsapp-web.js` Discussions tab: https://github.com/pedroslopez/whatsapp-web.js/discussions
+
+        Before submitting a new bug/issue, please check the links below to see if there is a solution or question posted there already:
+         - `whatsapp-web.js` Issues tab: https://github.com/pedroslopez/whatsapp-web.js/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc
+         - `whatsapp-web.js` closed issues tab: https://github.com/pedroslopez/whatsapp-web.js/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed
+         - `whatsapp-web.js` Discussions tab: https://github.com/pedroslopez/whatsapp-web.js/discussions
```
Comment (PurpShell) [inline .github/ISSUE_TEMPLATE/bug_report.yml:16]:
```suggestion
        inside of `whatsapp-web.js`'s Discord server: https://discord.gg/wyKybbF
        Before submitting a new bug/issue, please check the links below to see if there is a solution or question posted there already:
         - `whatsapp-web.js` Issues tab: https://github.com/pedroslopez/whatsapp-web.js/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc
         - `whatsapp-web.js` closed issues tab: https://github.com/pedroslopez/whatsapp-web.js/issues?q=is%3Aissue+sort%3Aupdated-desc+is%3Aclosed
         - `whatsapp-web.js` Discord server: https://discord.gg/wyKybbF
```

**Review by purpshell — COMMENTED** [review]
.

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 3 above)
Comment (PurpShell) [reply .github/ISSUE_TEMPLATE/bug_report.yml:16]: Discord is our weapon of choice 😎

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 2 above)
Comment (cliffordfajardo) [reply .github/ISSUE_TEMPLATE/bug_report.yml:10]: Sounds good 👌

**Comment by cliffordfajardo** [comment]
@PurpShell @pedroslopez  🙏

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      description: |
+        Which website or app were you using when the bug happened?
+        Note:
+        - Your bug will may get fixed much faster if we can run your code and it doesn't have dependencies other than the `whatsapp-web.js` npm package / dependency.
+        - To create a shareable code example you can use Stackblitz (https://stackblitz.com/). Please no localhost URLs.
+        - Please read these tips for providing a minimal example: https://stackoverflow.com/help/mcve.
+      placeholder: |
+        e.g. https://stackblitz.com/edit/...... OR Github Repo
+    validations:
+      required: true
```
Comment (pedroslopez) [inline .github/ISSUE_TEMPLATE/bug_report.yml:38]: Websites aren't really applicable here... Maybe this section should be replaced with a code sample

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      placeholder: |
+        You can drag your video or image files inside of this editor ↓
+  - type: textarea
+    id: platform
+    attributes:
+      label: Platform
+      value: |
+        - OS: [e.g. macOS, Windows, Linux]
+        - Browser: [e.g. Chrome, Safari, Firefox]
+        - Version: [e.g. 91.1]
```
Comment (pedroslopez) [inline .github/ISSUE_TEMPLATE/bug_report.yml:77]: All of these things that were in the previous template are super useful and I think should be kept:

```
**WhatsApp**
 - Account type [Standard / Business]:
 - Device OS [iOS / Android]:
 - WhatsApp Web version [run `await client.getWWebVersion()`]:

**Library**
 - Browser [Chrome / Chromium]:
 - NodeJS version (`node -v`):
 - npm or yarn version (`npm -v`):
 - whatsapp-web.js version:
 ```

**Review by pedroslopez — CHANGES_REQUESTED** [review]
Thank you for this! I think this would help improve issue quality a lot.

A few comments, but looks pretty good!

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 4 above)
Comment (cliffordfajardo) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Thanks - I've updated it; Below is a demo link: 
https://github.com/cliffordfajardo/whatsapp-web.js/blob/main/.github/ISSUE_TEMPLATE/bug_report.yml

File: .github/ISSUE_TEMPLATE/bug_report.yml (hunk 5 above)
Comment (cliffordfajardo) [reply .github/ISSUE_TEMPLATE/bug_report.yml]: Thanks - I've updated it; Below is a demo link: 
https://github.com/cliffordfajardo/whatsapp-web.js/blob/main/.github/ISSUE_TEMPLATE/bug_report.yml

**Comment by cliffordfajardo** [comment]
@pedroslopez & other core contributors friends 👥 
Feel free to make edit to my branch as the "Allow edits from maintainers" setting is enabled for my branch

Github now has Codespaces enabled for nearly all public repos on Github.com 🎉  
So you don't even need to clone &  run my branch on your local computer since Codespaces allows you to edit in the browser now .Apologies from my end with the back & forth changes 😅 
- https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests


My fork repo: https://github.com/cliffordfajardo/whatsapp-web.js

CleanShot 2022-02-28 at 02 30 52@2x

**Review by pedroslopez — APPROVED** [review]

**Pull request merged**